sync/handlers.py

```diff
@@ -195,27 +203,29 @@ def __call__(self, git_gecko, git_wpt, body):
         # the decision task is complete. This allows us to determine if a
         # try push should have the expected wpt tasks just by checking if
         # this is set
-        if result not in (tc.SUCCESS, tc.FAIL, tc.EXCEPTION):
-            logger.info("Decision task is not yet complete")
+        if state != "completed" or result == "superseded":
+            logger.info("Decision task is not yet complete, status %s" % result)
             return
 
         logger.info("Setting taskgroup id for try push %r to %s" % (try_push, task_id))
         try_push.taskgroup_id = task_id
 
-        failed_builds = len(try_push.failed_builds())
-        if result != tc.SUCCESS or failed_builds:
-            try_push.status = "complete"
-            try_push.infra_fail = True
+        if result in ("failure", "exception"):
             sync = try_push.sync(git_gecko, git_wpt)
             message = ("Decision task got status %s for task %s%s" %
                        (result, sha1, " PR %s" % sync.pr if sync and sync.pr else ""))
-            if failed_builds:
-                message = "The try push includes build failures. %s" % message
             logger.error(message)
-            if sync and sync.bug:
-                # TODO this is commenting too frequently on bugs
-                env.bz.comment(sync.bug,
-                               "Try push failed: decision task returned error")
+            taskgroup = tc.TaskGroup(task_id)
+            if len(taskgroup.tasks_by_id[task_id]) > 5:
+                try_push.status = "complete"
+                try_push.infra_fail = True
+                if sync and sync.bug:
+                    # TODO this is commenting too frequently on bugs
+                    env.bz.comment(sync.bug,
+                                   "Try push failed: decision task returned error")
+            else:
+                client = tc.TaskclusterClient()
+                client.retrigger(task_id)
```

Comment:
By the way, I tried calling retrigger on a decision task just now and our client doesn't have sufficient scopes. (Retriggering other kinds of tasks works fine.) Then, lucky me, I ran into a bug when trying to add the missing scopes to our client: https://bugzilla.mozilla.org/show_bug.cgi?id=1462459

In any case, related to my previous comment, maybe `taskcluster.Queue.rerun` is more appropriate here anyway. It's especially for failed/exception tasks and I think it would keep the same task_id.

Reply:
`rerunTask` is marked as deprecated in the docs, so I'm a little worried about using it. Is that what you meant or is there something else?